in-progress/6973-refactor-base-contract-interaction.md

```diff
@@ -0,0 +1,1092 @@
+  }
+
+  if (!builder.gasSettings) {
+    builder.setGasSettings(GasSettings.default());
+  }
+
+  await this.#ensureFunctionAbis(builder);
+
+  return this.#simulateInner(builder.build());
+}
+```
+
+
+### BaseWallet.prove
+
+```ts
+async prove(request: UserRequest): Promise<UserRequest> {
+  if (!request.executionResult) {
+    throw new Error('Execution result must be set before proving');
+  }
+  const builder = new UserRequestBuilder(request);
+  await this.#ensureFunctionAbis(builder);
+  const initRequest = builder.build();
+  const txExecutionRequest = await this.getTxExecutionRequest(initRequest);
+  const provenTx = await this.proveTx(txExecutionRequest, request.executionResult);
+  builder.setTx(provenTx);
+  return builder.build();
+}
+```
+
+### BaseWallet.send
+
+```ts
+async send(request: UserRequest): Promise<UserRequest> {
+  if (!request.tx) {
+    throw new Error('Tx must be set before sending');
+  }
+  if (!request.tx.proof || request.tx.proof.isEmpty()) {
+    throw new Error('Tx must be proven before sending');
+  }
+  const builder = new UserRequestBuilder(request);
+  await this.#ensureFunctionAbis(builder);
+  const initRequest = builder.build();
+  const txExecutionRequest = await this.getTxExecutionRequest();
+  const txHash = await this.sendTx(txExecutionRequest, request.tx);
+  return new SentTx(this.pxe, txHash, txExecutionRequest);
+}
+```
+
+### Dapp Funded Transactions
+
+
+#### A `TxExecutionRequestComponent` for the Dapp
+
+```ts
+export class DefaultDappInterface implements AccountInterface {
+  constructor(
+    private completeAddress: CompleteAddress,
+    private userAuthWitnessProvider: AuthWitnessProvider,
+    private dappEntrypointAddress: AztecAddress,
+    private chainId: number = DEFAULT_CHAIN_ID,
+    private version: number = DEFAULT_VERSION,
+  ) {}
+
+  async adaptTxExecutionRequest(
+    builder: TxExecutionRequestBuilder,
+    userRequest: UserRequest
+  ): Promise<void> {
+    if (builder.appFunctionCalls.length !== 1) {
+      throw new Error(`Expected exactly 1 function call, got ${calls.length}`);
+    }
+    if (builder.setupFunctionCalls.length !== 0) {
+      throw new Error(`Expected exactly 0 setup function calls, got ${calls.length}`);
+    }
+    const payload = EntrypointPayload.fromFunctionCalls(builder.appFunctionCalls);
+    const abi = this.getEntrypointAbi();
+
+    const entrypointPackedArgs = PackedValues.fromValues(encodeArguments(abi, [payload, this.completeAddress.address]));
+    const functionSelector = FunctionSelector.fromNameAndParameters(abi.name, abi.parameters);
+
+    const innerHash = computeInnerAuthWitHash([Fr.ZERO, functionSelector.toField(), entrypointPackedArgs.hash]);
+    const outerHash = computeOuterAuthWitHash(
+      this.dappEntrypointAddress,
+      new Fr(this.chainId),
+      new Fr(this.version),
+      innerHash,
+    );
+    const authWitness = await this.userAuthWitnessProvider.createAuthWit(outerHash);
```

Comment:
This is just `this.createAuthWit()`, can be written as that to just make it easier to see that they are the same functions though? Think the `createAuthWit` might just be using a variable that is not defined actually.

Reply:
The only difference between the two as the code presently exists (or at least when I wrote this) is that `this.createAuthWit` actually adds the result to the PXE which is not what is desired here.